Gems/Terrain/Code/Source/TerrainRenderer/TerrainFeatureProcessor.cpp

```diff
@@ -448,7 +466,35 @@ namespace Terrain
                 return AZ::RPI::PassFilterExecutionFlow::ContinueVisitingPasses;
             }
         );
+
+        // Clipmap compute pass
+        {
+            m_macroClipmapPass = nullptr;
+
+            AZ::Name passName = AZ::Name("TerrainMacroTextureComputePass");
+            AZ::RPI::PassFilter passFilter = AZ::RPI::PassFilter::CreateWithPassName(passName, pipeline);
+            AZ::RPI::Pass* pass = AZ::RPI::PassSystemInterface::Get()->FindFirstPass(passFilter);
+            if (pass)
+            {
+                auto* clipmapPass = azrtti_cast<Terrain::TerrainMacroTextureComputePass*>(pass);
+                m_macroClipmapPass = clipmapPass;
+                m_macroClipmapPass->SetFeatureProcessor(this);
+            }
+            else
+            {
+                AZ_Error(
+                    "TerrainFeatureProcessor", false,
+                    "%s does not exist in this pipeline. Check your game project's .pass assets.", passName.GetCStr());
+            }
+        }
     }
-    
 
+    void TerrainFeatureProcessor::EnablePasses()
+    {
+        if (m_macroClipmapPass)
+        {
+            // We will not enable it until clipmap replaces the current implementation.
+            m_macroClipmapPass->SetEnabled(false);
```

Comment:
I understand not enabling the passes for now, but having a function called "EnablePasses" which actually disables the pass is pretty confusing. It seems like this is something that could just be controlled from the pipeline json - you can set passes to enabled / disabled there, and then there's no need to have this function at all. Eventually I think you'll want to be able to switch modes between clipmap and forward, but that'll involve a little more work than just turning passes on and off.

Reply:
This is not necessary but I added it for some flexibility. The pipeline json has already disabled the pass. But we are still blocked by the ability to add pass requests dynamically to the pipeline, when enabling a gem. I am thinking have them disabled in project's MainPipeline and enabling through the gem code.
I'll remove it for now.